src/lib.rs

```diff
@@ -4,16 +4,21 @@
 
 #![deny(missing_docs)]
 
+mod address_allocator;
 /// Allocation engine used by address allocator.
-pub mod allocation_engine;
+mod allocation_engine;
 mod id_allocator;
 
+pub use crate::address_allocator::AddressAllocator;
+use crate::allocation_engine::NodeState;
 pub use crate::id_allocator::IdAllocator;
```

Comment:
In the IdAllocator doc comments there are some comments about the implementation. Those should be hidden. We can open a separate issue for that. I've spotted them by doing a `cargo doc --open`. Here are the comments that I am referring to: https://github.com/rust-vmm/vm-allocator/blob/ac13f051feef3c65ec76213770c87039cfd83871/src/id_allocator.rs#L14. These should be written with `//`, and we should have something more high level, concerning the user in the `///` type of comments.

Reply:
I will create a new PR afterwards to update the documentation, comments, some tests and so on. Will create a separate issue and address these there. :D